### PR TITLE
Propagate BindingContent to Border children

### LINKED ISSUE #2693: [Bug] Elements wrapped with Border inside of CollectionView don't render.
### Description

If you have a CollectionView and wrap the elements of its ItemTemplate with a Border, none of the contents seem to render at all.

image

image

In my example, without the border, the contents do render correctly.

### LINKED ISSUE #3304: [Bug] Border not propagating bindings
### Description

WeatherTwentyOne has a Favorites page with a `CollectionView`. I have 2 template designs. The one that uses `Frame` works great.

Screenshot_1636408942

The one that uses `Border` does not display the binding content. Using Android's layout inspector I have confirmed the UI elements are present and the bound content is missing.

Screenshot_1636409154

### Basic Information

- Version with issue: pre10

### Reproduction Link

https://github.com/davidortinau/WeatherTwentyOne

```
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:v="clr-namespace:WeatherTwentyOne.Views"
             xmlns:m="clr-namespace:WeatherTwentyOne.Models"
             xmlns:c="clr-namespace:WeatherClient2021"
             xmlns:ios="clr-namespace:Microsoft.Maui.Controls.PlatformConfiguration.iOSSpecific;assembly=Microsoft.Maui.Controls"
             ios:Page.UseSafeArea="True"
             Title="Favorites"
             x:Class="WeatherTwentyOne.Pages.FavoritesPage"
             >
    <ContentPage.Resources>
        <DataTemplate x:Key="FavTemplate">
            <Frame
                Padding="0"
                CornerRadius="20"
                HasShadow="False"
                x:DataType="c:Location"
                BackgroundColor="{StaticResource Background_Mid}">
                <GridLayout
                                RowDefinitions="42,40,*"
                                Padding="20">
                    <Image
                                    Source="{Binding Icon}"
                                    WidthRequest="36"
                                    HeightRequest="36"
                                    VerticalOptions="Start"
                                    HorizontalOptions="End"/>

                    <Label Text="{Binding Value}" class="LargeTitle"
                                           />

                    <VerticalStackLayout
                                    Spacing="0"
                                    GridLayout.RowSpan="3" 
                                    VerticalOptions="Center">
                        <Label Text="{Binding Name}" class="Subhead" />
                        <Label Text="{Binding WeatherStation}" class="SubContent"/>
                    </VerticalStackLayout>

                    <HorizontalStackLayout
                                GridLayout.Row="2"
                                Spacing="10"
                                VerticalOptions="End">
                        <Image Source="solid_umbrella.png" VerticalOptions="Center"
                                       WidthRequest="20" HeightRequest="20"/>
                        <Label Text="13%" class="Small" VerticalOptions="Center"/>
                    </HorizontalStackLayout>

                    <HorizontalStackLayout
                                    GridLayout.Row="2"
                                    Spacing="10"
                                    HorizontalOptions="End"
                                    VerticalOptions="End">
                        <Image Source="solid_humidity.png" VerticalOptions="Center"
                                       WidthRequest="20" HeightRequest="20"/>
                        <Label Text="45%" class="Small" VerticalOptions="Center"/>
                    </HorizontalStackLayout>
                </GridLayout>
            </Frame>

            

        </DataTemplate>

        <DataTemplate x:Key="FavBorderTemplate">
            <Border
                            Margin="{OnIdiom Phone=15,Default=25}"
                            Padding="0"
                            HorizontalOptions="Start"
                            HeightRequest="154"
                            WidthRequest="154"
                            Stroke="Transparent"
                            StrokeThickness="1"
                            BackgroundColor="{StaticResource Background_Mid}"
                            x:DataType="c:Location"
                            >
                <Border.StrokeShape>
                    <RoundRectangle CornerRadius="40"/>
                </Border.StrokeShape>
                <GridLayout
                                RowDefinitions="42,40,*"
                                Padding="20">
                    <Image
                                    Source="{Binding Icon}"
                                    WidthRequest="36"
                                    HeightRequest="36"
                                    VerticalOptions="Start"
                                    HorizontalOptions="End"/>

                    <Label Text="{Binding Value}" class="LargeTitle"
                                           />

                    <VerticalStackLayout
                                    Spacing="0"
                                    GridLayout.RowSpan="3" 
                                    VerticalOptions="Center">
                        <Label Text="{Binding Name}" class="Subhead" />
                        <Label Text="{Binding WeatherStation}" class="SubContent"/>
                    </VerticalStackLayout>

                    <HorizontalStackLayout
                                GridLayout.Row="2"
                                Spacing="10"
                                VerticalOptions="End">
                        <Image Source="solid_umbrella.png" VerticalOptions="Center"
                                       WidthRequest="20" HeightRequest="20"/>
                        <Label Text="13%" class="Small" VerticalOptions="Center"/>
                    </HorizontalStackLayout>

                    <HorizontalStackLayout
                                    GridLayout.Row="2"
                                    Spacing="10"
                                    HorizontalOptions="End"
                                    VerticalOptions="End">
                        <Image Source="solid_humidity.png" VerticalOptions="Center"
                                       WidthRequest="20" HeightRequest="20"/>
                        <Label Text="45%" class="Small" VerticalOptions="Center"/>
                    </HorizontalStackLayout>
                </GridLayout>
            </Border>
        </DataTemplate>
    </ContentPage.Resources>
    
    
        <!-- Main Section -->
        <ScrollView>
            <VerticalStackLayout>

                <!-- Search Box -->
                <HorizontalStackLayout
                    VerticalOptions="Start"
                    Spacing="18"
                    Padding="{OnIdiom Phone='15,15,15,15', Default='25,25,25,0'}"
                    >
                    <Image
                        Source="search_icon.png"
                        VerticalOptions="Center"
                        HeightRequest="22"
                        WidthRequest="22"/>
                    <Label
                        Text="Search"
                        TextColor="#637989"
                        FontSize="18"
                        WidthRequest="240"
                        VerticalOptions="Center"/>
                </HorizontalStackLayout>

                <!-- Actual Content -->
            <CollectionView
                Margin="8,0"
                ItemsSource="{Binding Favorites}"
                ItemTemplate="{StaticResource FavBorderTemplate}">
                <CollectionView.ItemsLayout>
                    <GridItemsLayout Orientation="Vertical"
                                     Span="2"
                                     VerticalItemSpacing="8"
                                     HorizontalItemSpacing="8"/>
                </CollectionView.ItemsLayout>
                <CollectionView.FooterTemplate>
                    <DataTemplate>
                        <Border
                            HeightRequest="132"
                            WidthRequest="132"
                            HorizontalOptions="Start"
                            Margin="{OnIdiom Phone=15,Default=25}"
                            Stroke="Transparent"
                            StrokeThickness="1"
                            BackgroundColor="{StaticResource Background_Mid}">
                            <Border.StrokeShape>
                                <RoundRectangle CornerRadius="60"/>
                            </Border.StrokeShape>
                            <ImageButton
                                BackgroundColor="Transparent"
                                Source="add_icon.png"
                                VerticalOptions="Center" HorizontalOptions="Center"/>
                        </Border>
                    </DataTemplate>
                </CollectionView.FooterTemplate>
            </CollectionView>

        </VerticalStackLayout>
        </ScrollView>

</ContentPage>
```

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Propagate BindingContent to Border children.

- fixes #3304 
- fixes #2693

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)

#### Does this PR touch anything that might affect accessibility?
- No

### CODE REVIEW

File: src/Controls/src/Core/Border.cs [hunk 1]
```diff
@@ -145,6 +145,18 @@ public float[] StrokeDashPattern
 
 		IView IContentView.PresentedContent => Content;
 
+		protected override void OnBindingContextChanged()
+		{
+			base.OnBindingContextChanged();
+
+			BindableObject content = Content as BindableObject;
+
+			if (content != null)
```
Comment (AmrAlSayed0) [inline src/Controls/src/Core/Border.cs:154]: Is this accounted for somewhere? maybe the base `BindableObject`.
```csharp
var border = Border();

var bindingContext = new object();
border.BindingContext = bindingContext;

var content = new ContentView();
border.Content = content;

Assert.True(content.BindingContext == bindingContext);
Assert.True(content.BindingContext == border.BindingContext);
```

**Review by mattleibow — DISMISSED** [review]

**Comment by mattleibow** [comment]
@PureWeen is this a by-product of the border content not being added to the logical children? https://github.com/dotnet/maui/pull/3291

File: src/Controls/src/Core/Border.cs (hunk 1 above)
Comment (mattleibow) [reply src/Controls/src/Core/Border.cs:154]: I believe it is this test: https://github.com/dotnet/maui/pull/3315/files#diff-a186d7040a43a5d73d4827e554cf6c9ce961506def6072d802032628d4d14d5fR19

**Comment by mattleibow** [comment]
Seems the first PR fixed this and the tests pass now.

**Review by mattleibow — APPROVED** [review]
Approving this mainly for the updated xaml and new tests. The #3291 PR actually used the base logic for children and that now propagates the context - as well as the styles and things.

**Pull request merged**